pulser-core/pulser/waveforms.py

```diff
@@ -466,8 +492,10 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"CustomWaveform({self.duration} ns, {self.samples!r})"
 
-    def __mul__(self, other: float) -> CustomWaveform:
-        return CustomWaveform(self._samples * float(other))
+    def __mul__(self, other: float | ArrayLike) -> CustomWaveform:
+        return CustomWaveform(
+            self._samples * pm.AbstractArray(other, dtype=float)
```

Comment:
Are we sure this dtype work with torch tensors ? 😮

Reply:
Yes :)

Reply:
Then I flatly resolve this conservation.